### PR TITLE
SRCH-3994 Backend data prep for spelling suggestions

### PR DESCRIPTION
**Pull request by lorawoodford**
## Summary
- Prepares backend spelling suggestion data for future integration into the redesigned SERP
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [ ] ⚠️ Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
Note: These changes raise a new Code Climate warning as I've added a fifth argument to the `search_results_layout` method in `react_helper`.  I propose we just ignore this warning for now (it shouldn't fire again once ignored this time) as Code Climate passing is not required by our branch protection rules, and because we know that (hopefully) the way `react_helper` is implemented won't persist for long (see: https://github.com/GSA/search-gov/pull/1329/files#r1373775372).  Happy to discuss further.
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/helpers/react_helper.rb [hunk 1]
```diff
@@ -62,6 +63,21 @@ def news_label(search)
...
+  def spelling_text(search, search_options)
+    return if search.spelling_suggestion.blank?
+
+    spelling_suggestion_links(search, search_options) do |suggested_query, suggested_url, original_url|
+      {
+        suggested: t(:showing_results_for,
+                     corrected_query: link_to(suggested_query,
+                                              suggested_url)),
+        original: t(:search_instead_for,
+                    original_query: link_to(search.query,
```
Comment (MyNameIsMissing) [inline app/helpers/react_helper.rb:75]: do we care about word_otherword vs word.otherword?  Ruby convention is underscore right?

**Review by MyNameIsMissing — APPROVED** [review]
LGTM.  One small question for my own edification.

File: app/helpers/react_helper.rb (hunk 1 above)
Comment (lorawoodford) [reply app/helpers/react_helper.rb:75]: I'm not entirely following, but if you're asking why `search.query` that's necessary as query is an attribute of the search object and `search.query` is how we must access it.  Is that what you're curious about?

On the other hand, `suggested_query`, `suggested_url`, `original_url` are variables being returned via the preexisting `spelling_suggestion_links` method defined here: https://github.com/GSA/search-gov/blob/main/app/helpers/spelling_suggestions_helper.rb#L2. (As an aside, we probably wouldn't write a method like that these days that yields multiple values, but it's already there, it works, it's covered by multiple tests, and it's likely unwise to refactor it as part of this effort, so I've just left it be.)

Comment (MyNameIsMissing) [reply app/helpers/react_helper.rb:75]: Yes that explains.  Thanks!

**Comment by lorawoodford** [comment]
@yogesh27 Updated per our discussion.

**Comment by lorawoodford** [comment]
@yogesh27 See my comment in the PR above... we can ignore this one for the time being (passing code climate is not a requirement to merge) and on subsequent PR there should not be new failures.

**Review by yogesh27 — APPROVED** [review]
LGTM

**Pull request merged**